### PR TITLE
CT: Add source filename in error output when possible

### PR DESCRIPTION
**Pull request by essen**
PoC for https://github.com/erlang/otp/issues/10260

I figured that I would get the filename from the same place exceptions already do since it's the most appropriate and there's no need to tweak the filename. With this filename I could then have GitHub annotate PRs with the exact failure location.

In the shell it looks like this:

```
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
compress_SUITE:gzip_accept_encoding_malformed (test/compress_SUITE.erl) failed on line 75
Reason: {badmatch,2}
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
```

Or

```
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
cowboy_test:gun_open (test/cowboy_test.erl) failed on line 184
Reason: {badmatch,{error,{no_quicer,"Cowboy must be compiled with en...}
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
```

I haven't added tests, let's see if this is an approach that would be OK first.

### CODE REVIEW

**Comment by michalmuskala** [comment]
With the same logic as in https://github.com/erlang/otp/pull/9468, it would likely be better to have the format as `path/to/file:LINE` - many tools recognise that and make it "clickable". This would also align the format with compiler errors and stack trace printing (since OTP 28).

cc @TheGeorge

**Comment by essen** [comment]
Sure, we could do that when we have a filename. What if we don't, do we keep the error message as it is now?

**Comment by TheGeorge** [comment]
We could emulate this via <module_name>.erl:<line> or keep the existing format

**Comment by essen** [comment]
```
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
cowboy_test:gun_open at test/cowboy_test.erl:184 failed
Reason: {badmatch,{error,{no_quicer,"Cowboy must be compiled with en...}
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
```

**Comment by essen** [comment]
Hello, any chance this can be reviewed/merged? Also #10277.

**Comment by Mikaka27** [comment]
Hi, I think we can go with this solution until we have time to untangle the common_test to keep source file information. I will check the other PR next.

You said you didn't added tests yet, could you add them?

**Comment by essen** [comment]
Hello! Thanks for reviewing.

In the normal case many tests already go through this code. We probably don't want to test the exact string being output.

The edge case covered by the `else` clause can be interesting to test on the other hand, to at least know we are not breaking when a file doesn't include abstract code. Will do that one soon.

**Comment by essen** [comment]
What do you think about 86660db? I basically took the misc_errors test and changed it to compile without debug_info (and with deterministic to go one step further). The compiled SUITE file is removed and recompiled in the misc_errors test so it runs as before. I have confirmed manually that it works as intended since the output is the same as normal in this case.

File: lib/common_test/test/ct_error_SUITE.erl [hunk 1]
```diff
@@ -156,6 +156,37 @@ no_compile(Config) when is_list(Config) ->
...
+%%%-----------------------------------------------------------------
+%%%
+no_debug_info(Config) when is_list(Config) ->
+
+    DataDir = ?config(data_dir, Config),
+
+    TestDir = filename:join(DataDir, "error/test/"),
+    ModFile = filename:join(TestDir, "misc_error_1_SUITE.erl"),
+    ct:pal("Compiling ~p: ~p",
+	   [ModFile,compile:file(ModFile,[{outdir,TestDir},deterministic])]),
```
Comment (Mikaka27) [inline lib/common_test/test/ct_error_SUITE.erl]: I think it would be a good idea to check return value from compile:file here, so that we're sure we're testing on module compiled with those options. Otherwise the module will be compiled automatically with debug_info. At least that's what happens in my testing.

File: lib/common_test/test/ct_error_SUITE.erl [hunk 2]
```diff
@@ -156,6 +156,37 @@ no_compile(Config) when is_list(Config) ->
...
 
+%%%-----------------------------------------------------------------
+%%%
+no_debug_info(Config) when is_list(Config) ->
+
+    DataDir = ?config(data_dir, Config),
+
+    TestDir = filename:join(DataDir, "error/test/"),
+    ModFile = filename:join(TestDir, "misc_error_1_SUITE.erl"),
+    ct:pal("Compiling ~p: ~p",
```
Comment (Mikaka27) [inline lib/common_test/test/ct_error_SUITE.erl]: Please change this from pal to log. There was some effort already to reduce prints in console during test execution. If you don't mind you could change pal to log in init_per_suite function as well :)

File: lib/common_test/test/ct_error_SUITE.erl [hunk 3]
```diff
@@ -156,6 +156,37 @@ no_compile(Config) when is_list(Config) ->
...
+
+    ct_test_support:log_events(misc_errors,
+			       reformat(Events, ?eh),
+			       ?config(priv_dir, Config),
+			       Opts),
+
+    TestEvents = events_to_check(misc_errors),
+    ok = ct_test_support:verify_events(TestEvents, Events, Config)
+    after
+        ct:pal("Removing ~p", [ModFile]),
```
Comment (Mikaka27) [inline lib/common_test/test/ct_error_SUITE.erl]: Same as above, change from pal to log.

**Comment by essen** [comment]
All done.

**Comment by Mikaka27** [comment]
Thanks, I will run this through CI overnight, and merge tomorrow if there are no problems :)
